### PR TITLE
[risk=no] Fmh generate skip counts

### PR DESCRIPTION
**Pull request by SrushtiGangireddy**
1. Separating survey counts generation queries into different script.
2. We now have to group counts of family medical history answers by condition and by family member.
3. The structure of survey in observation table do not ease straight forward pull.
4. We have intermediate metadata tables that help this pull.
5. Also, the skip codes pull needs different query and grouping (+ that in this pr)

### CODE REVIEW

**Comment by SrushtiGangireddy** [comment]
generate-survey-counts.sh, generate-fmh-condition-counts.sh, generate-fmh-family-member-counts.sh are new files that have queries just pulled out from run-achilles-queries.sh

File: public-api/db-cdr/generate-cdr/generate-fmh-condition-counts.sh [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+,\`${WORKBENCH_PROJECT}.${WORKBENCH_DATASET}.v_full_observation\` ob JOIN UNNEST(SPLIT(fmh_concepts_to_count,',')) as concepts on cast(ob.value_source_concept_id as string)=concepts
+join \`${BQ_PROJECT}.${BQ_DATASET}.concept\` c on c.concept_id=cast(concepts as int64)
+join \`${WORKBENCH_PROJECT}.${WORKBENCH_DATASET}.survey_age_stratum\` sa on sa.observation_id=ob.observation_id
+where exists
+(select * from \`${WORKBENCH_PROJECT}.${WORKBENCH_DATASET}.v_full_observation\`
+where questionnaire_response_id=ob.questionnaire_response_id and
+observation_source_concept_id=43528652 and value_source_concept_id in (43529842, 43528385))
+group by 4,5,6"
+
+# Calculate skip count of each condition
```
Comment (SrushtiGangireddy) [inline public-api/db-cdr/generate-cdr/generate-fmh-condition-counts.sh:220]: This is new query that calculates skip count of each condition by counting sum of mother condition skip + father condition skip + sibling condition skip + ....

File: public-api/db-cdr/generate-cdr/generate-fmh-condition-counts.sh [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+(select person_id,fmh.observation_source_concept_id as question_concept, string_agg(distinct cast(ob.observation_source_concept_id as string), ',') as questions_skipped from
+\`${WORKBENCH_PROJECT}.${WORKBENCH_DATASET}.fmh_conditions_member_metadata\` fmh, \`${WORKBENCH_PROJECT}.${WORKBENCH_DATASET}.fmh_conditions_member_metadata\` ob
+join UNNEST(SPLIT(concepts_to_count,',')) as concepts on ob.observation_source_concept_id=cast(concepts as int64)
+where value_source_concept_id=903096 group by 1,2)
+select 0 as id, 3110 as analysis_id,'43528698' as stratum_1,
+cast(question_concept as string) as stratum_2, '903096' as stratum_3,'Skip' as stratum_4, '2' as stratum_5, count(person_id) as count_value, 0 as source_count_value
+from a where ARRAY_LENGTH(SPLIT(questions_skipped))=6
+group by 4"
+
+bq --quiet --project=$BQ_PROJECT query --nouse_legacy_sql \
```
Comment (SrushtiGangireddy) [inline public-api/db-cdr/generate-cdr/generate-fmh-condition-counts.sh:241]: breakdown for same query

File: public-api/db-cdr/generate-cdr/generate-fmh-condition-counts.sh [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+\`${WORKBENCH_PROJECT}.${WORKBENCH_DATASET}.fmh_conditions_member_metadata\` fmh, \`${WORKBENCH_PROJECT}.${WORKBENCH_DATASET}.v_full_observation\` ob
+join UNNEST(SPLIT(concepts_to_count,',')) as concepts on ob.observation_source_concept_id=cast(concepts as int64)
+join \`${WORKBENCH_PROJECT}.${WORKBENCH_DATASET}.v_person\` p on ob.person_id=p.person_id
+where value_source_concept_id=903096 group by 1,2,3)
+select 0 as id, 3111 as analysis_id,'43528698' as stratum_1,
+cast(question_concept as string) as stratum_2, '903096' as stratum_3,'Skip' as stratum_4, cast(gender as string) as stratum_5, '2' as stratum_6, count(person_id) as count_value, 0 as source_count_value
+from a where ARRAY_LENGTH(SPLIT(questions_skipped))=6
+group by 4,7"
+
+bq --quiet --project=$BQ_PROJECT query --nouse_legacy_sql \
```
Comment (SrushtiGangireddy) [inline public-api/db-cdr/generate-cdr/generate-fmh-condition-counts.sh:260]: age breakdown for same query

File: public-api/db-cdr/generate-cdr/generate-fmh-family-member-counts.sh [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+join \`${WORKBENCH_PROJECT}.${WORKBENCH_DATASET}.survey_age_stratum\` sa on sa.observation_id=ob.observation_id
+join UNNEST(SPLIT(concepts_to_count,',')) as concepts
+on ob.observation_source_concept_id=cast(concepts as int64)
+where exists
+(select * from \`${WORKBENCH_PROJECT}.${WORKBENCH_DATASET}.v_full_observation\`
+where questionnaire_response_id=ob.questionnaire_response_id and
+observation_source_concept_id=43528652 and value_source_concept_id in (43529842, 43528385))
+group by 4,5,6"
+
+# Calculate skip, do not know, prefer not to answer count of each family member history
```
Comment (SrushtiGangireddy) [inline public-api/db-cdr/generate-cdr/generate-fmh-family-member-counts.sh:231]: Calculating mother skip by counting mother cancer skip + mother respiratory condition skip + mother kidney condition skip + ....

File: public-api/db-cdr/generate-cdr/generate-fmh-family-member-counts.sh [hunk 5]
```diff
@@ -0,0 +1,303 @@
...
+and ob.observation_source_concept_id not in (43529658, 43529656, 43529659, 43529655, 43529660, 43529657)
+group by 1,2,3,5)
+select 0 as id, 3110 as analysis_id, '43528698' as stratum_1,
+cast(concept as string) as stratum_2, cast(value_source_concept_id as string) as stratum_3, c.concept_name as stratum_4, '3' as stratum_5,
+cast(concept as string) as stratum_6, count(distinct person_id) as count_value, 0 as source_count_value
+from a
+where ARRAY_LENGTH(SPLIT(questions_skipped))=11
+group by 4,5,6,8"
+
+# Biological sex breakdown
```
Comment (SrushtiGangireddy) [inline public-api/db-cdr/generate-cdr/generate-fmh-family-member-counts.sh:258]: biological sex breakdown of same query

File: public-api/db-cdr/generate-cdr/generate-fmh-family-member-counts.sh [hunk 6]
```diff
@@ -0,0 +1,303 @@
...
+observation_source_concept_id=43528652 and value_source_concept_id in (43529842, 43528385))
+group by 1,2,3,4,6)
+select 0 as id, 3111 as analysis_id, '43528698' as stratum_1,
+cast(concept as string) as stratum_2, cast(value_source_concept_id as string) as stratum_3, c.concept_name as stratum_4, gender as stratum_5,
+cast(concept as string) as stratum_6, count(distinct person_id) as count_value, 0 as source_count_value
+from a
+where ARRAY_LENGTH(SPLIT(questions_skipped))=11
+group by 4,5,6,7,8"
+
+# Age breakdown
```
Comment (SrushtiGangireddy) [inline public-api/db-cdr/generate-cdr/generate-fmh-family-member-counts.sh:284]: age breakdown of same query

**Review by freemabd — APPROVED** [review]
Thanks

**Review by freemabd — APPROVED** [review]

**Pull request merged**